From f4a12e79aa7fe603167386e1f77068ce89fc4d23 Mon Sep 17 00:00:00 2001
From: Benjamin Baratte <benjamin.baratte@st.com>
Date: Thu, 9 Feb 2023 10:35:01 +0100
Subject: [PATCH 04/13] Allow SE key to use key vendor id within PSA crypto

Signed-off-by: Benjamin Baratte <benjamin.baratte@st.com>
---
 core/psa_crypto.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/psa_crypto.c b/core/psa_crypto.c
index de131f58d..5d09f0585 100644
--- a/core/psa_crypto.c
+++ b/core/psa_crypto.c
@@ -1765,7 +1765,11 @@ static psa_status_t psa_validate_key_attributes(const psa_key_attributes_t *attr
             return PSA_ERROR_INVALID_ARGUMENT;
         }
     } else {
+#ifdef MBEDTLS_PSA_CRYPTO_SE_C
+        if (!psa_is_valid_key_id(psa_get_key_id(attributes), 1)) {
+#else
         if (!psa_is_valid_key_id(psa_get_key_id(attributes), 0)) {
+#endif
             return PSA_ERROR_INVALID_ARGUMENT;
         }
     }
-- 
2.34.1

