From f93b18dcbb6ed16a437a843a266477008e7cc9cc Mon Sep 17 00:00:00 2001
From: Waleed Elmelegy <waleed.elmelegy@arm.com>
Date: Thu, 14 Aug 2025 14:11:58 +0000
Subject: [PATCH 10/10] Add TF-M Opaque  key entry points PSA Cipher API

Signed-off-by: Waleed Elmelegy <waleed.elmelegy@arm.com>
---
 library/psa_crypto_driver_wrappers.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/library/psa_crypto_driver_wrappers.h b/library/psa_crypto_driver_wrappers.h
index 4ff123a92b..a3be698ca2 100644
--- a/library/psa_crypto_driver_wrappers.h
+++ b/library/psa_crypto_driver_wrappers.h
@@ -18,6 +18,9 @@
 #include "psa_crypto_mac.h"
 #include "psa_crypto_pake.h"
 #include "psa_crypto_rsa.h"
+#if defined(CC3XX_CRYPTO_OPAQUE_KEYS)
+#include "cc3xx_opaque_keys.h"
+#endif
 
 #include "mbedtls/platform.h"
 #include "mbedtls/constant_time.h"
@@ -1493,6 +1496,9 @@ static inline psa_status_t psa_driver_wrapper_cipher_decrypt_setup(
     switch( location )
     {
         case PSA_KEY_LOCATION_LOCAL_STORAGE:
+#if defined(CC3XX_CRYPTO_OPAQUE_KEYS)
+        case CC3XX_OPAQUE_KEY_LOCATION:
+#endif
 #if defined(PSA_CRYPTO_DRIVER_TFM_BUILTIN_KEY_LOADER)
         case TFM_BUILTIN_KEY_LOADER_KEY_LOCATION:
 #endif /* PSA_CRYPTO_DRIVER_TFM_BUILTIN_KEY_LOADER */
-- 
2.25.1

